custom/icds_reports/utils/aggregation_helpers/distributed/awc_location.py

```diff
@@ -51,7 +105,15 @@ def drop_table_query(self):
             DELETE FROM "{tablename}";
         """.format(tablename=self.base_tablename)
 
-    def aggregate_query(self):
+    def aggregate_to_temporary_table(self, cursor, csv_file):
+        cursor.copy_from(csv_file, 'temp_awc_location')
+
+    def create_temporary_table_query(self):
+        return "CREATE TEMPORARY TABLE \"temp_awc_location\" AS SELECT * FROM \"{tablename}\"".format(
+            tablename=self.base_tablename,
+        )
+
+    def move_data_to_real_table(self):
```

Comment:
what about dropping the table and renaming the 'temp' table?

Reply:
I could be wrong about this, but I though that

```
DROP TABLE ...
RENAME TABLE ...
```

would cause a lock in such a way that all queries would be held, but 

```
DELETE FROM 
INSERT
```

would be more inefficient because it would keep two copies of data in the table (as dead tuples), but it could happen concurrently with queries. 

I can look into this more if I have some time, but may not prioritize for release